discovery_engine_core/core/src/stack/ops/trusted.rs

```diff
@@ -41,7 +41,7 @@ use super::{common::request_min_new_items, NewItemsError, Ops};
 
 /// Stack operations customized for trusted news.
 pub(crate) struct TrustedNews {
-    client: Arc<Client>,
+    client: Arc<dyn HeadlinesProvider>,
```

Comment:
>  is there a specific reason

Yes, it's the whole point of this PR, allowing use to at runtime deciding which provider implementation to use based on the configuration.

Except that in this PR it will always fixed use newscatcher as the multiplexing code is in a follow up PR. But in this PR we already should unify over the different providers.

Also the performance difference is so small that it's likely not even noticeable, I mean that one vtable call for a request which then fetches something from the internet...